pca9548a/pca9685a.go

```diff
@@ -0,0 +1,87 @@
+// Package pca9548a provides a driver for the PCA9548A 8-channel I2C-bus.
+//
+// Datasheet:
+// https://www.nxp.com/docs/en/data-sheet/PCA9548A.pdf
+package pca9548a // import "tinygo.org/x/drivers/pca9548a"
+
+import (
+	"tinygo.org/x/drivers"
+)
+
+// Device is a handle to the PCA9548A device given an address
+type Device struct {
+	addr uint8
+	bus  drivers.I2C
+	buf  [4]byte
+}
+
+// New creates a new instance of a PCA9548A device. It performs
+// no IO on the i2c bus.
+func New(bus drivers.I2C, addr uint8) Device {
+	return Device{
+		bus:  bus,
+		addr: addr,
+	}
+}
+
+// Connected returns whether a PCA9548A has been found.
+// It does a "who am I" request and checks the response.
+func (d *Device) IsConnected() bool {
+	d.SetPortState(0xA5)
+	response := d.GetPortState()
+	d.SetPortState(0x00)
+	return response == 0xA5
+}
+
+// SetPort enables the given port to send data.
+func (d *Device) SetPort(portNumber byte) {
+	portValue := uint8(0)
+	if portNumber <= 7 {
+		portValue = 1 << portNumber
+	}
+	d.bus.Tx(uint16(d.addr), []byte{portValue}, nil)
+}
+
+// GetPort gets the first port enabled.
+func (d *Device) GetPort() byte {
+	portBits := d.GetPortState()
+	for i := uint8(0); i < 8; i++ {
+		if (portBits & (1 << i)) != 0x00 {
+			return i
+		}
+	}
+	return 99
```

Comment:
`99`?

Reply:
I wasn't sure what value return ... that should be in case no port is enable. 0 is the first port and can not be used. Any value >= 16 is an invalid port and could be used.

Other idea is to return (port byte, err error) but again, a value must be used when error :thinking:

Reply:
> Any value >= 16 is an invalid port and could be used.

In that case how about:

```go
const InvalidPort = 0xff
```

Reply:
added, thanks